spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/SpringSecurityAuditorAware.java

```diff
@@ -41,8 +43,8 @@ public SpringSecurityAuditorAware(SecurityStateBean securityStateBean) {
 		final boolean authenticationEnabled = securityStateBean.isAuthenticationEnabled();
 		if (authenticationEnabled && SecurityContextHolder.getContext() != null) {
 			final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
-			if (!(authentication instanceof AnonymousAuthenticationToken)) {
-				return Optional.of(authentication.getName());
+			if (authentication != null && !(authentication instanceof AnonymousAuthenticationToken)) {
```

Comment:
I believe this null check is redundant as `null instanceof Foo)` is always `false`.

Reply:
Ah there's the `!`, which then needs a null check.

Reply:
Touche' ..... LOL great catch